### PR TITLE
fix(cli): separate experimental rules in `--debug`

### PR DESCRIPTION
**Pull request by brandonspark**
## What:
Currently, the `semgrep --debug` output displays a list of all the rules that have been run. Experimental rules are not displayed differently, however, which can be confusing to people who are running experimental rules, but are unaware.

## Why:
We should fix this so `semgrep` is more clear about experimental rules and when they are run.

## How:
Separated out experimental rules in the `--debug` output. Added some tests.

## Test plan:
`make test`

Closes PA-2159.

PR checklist:

- [X] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [X] Tests included or PR comment includes a reproducible test plan
- [X] Documentation is up-to-date
- [X] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [X] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by brendon** [comment]
Wrong @brendon ;)

File: cli/tests/e2e/snapshots/test_check/test_max_memory/error.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@ running 1 rules from 1 config rules/long.yaml_0
 No .semgrepignore found. Using default .semgrepignore rules. See the docs for the list of default ignores: https://semgrep.dev/docs/cli-usage/#ignoring-files
 Rules:
 - rules.forcetimeout
+Experimental Rules:
```
Comment (brendongo) [inline cli/tests/e2e/snapshots/test_check/test_max_memory/error.txt]: Maybe don't print this out if there are no experimental rules?

**Review by brendongo — CHANGES_REQUESTED** [review]
see above

File: cli/tests/e2e/snapshots/test_check/test_max_memory/error.txt (hunk 1 above)
Comment (brandonspark) [reply cli/tests/e2e/snapshots/test_check/test_max_memory/error.txt]: fair. i was giong off the fact taht `Rules` is printed even if there are none, but i realize that this is probably a much more likely occurrence

**Comment by brandonspark** [comment]
@brendongo any thoughts?

**Review by brendongo — APPROVED** [review]

**Pull request merged**